ms/MSOper/MSMetaData.cc

```diff
@@ -1103,6 +1102,32 @@ uInt MSMetaData::nFields() const {
     return nFields;
 }
 
+SHARED_PTR<std::set<Int> > MSMetaData::_getEphemFieldIDs() const {
+    // responsible for setting _ephemFields
+    if (_ephemFields) {
+        return _ephemFields;
+    }
+    ROMSFieldColumns msfc(_ms->field());
+    ROScalarColumn<Int> ephemCol = msfc.ephemerisId();
+    _ephemFields.reset(new std::set<Int>());
+    if (ephemCol.isNull()) {
+        return _ephemFields;
+    }
+    Vector<Int> colData = ephemCol.getColumn();
+    if (! anyTrue(colData >= 0)) {
+        return _ephemFields;
+    }
```

Comment:
this fastpath does not seem necessary, it might even be slower than the other loop as it creates a temporary.